tests/v2/validation/charts/backup_restore/backup_restore_test.go

```diff
@@ -0,0 +1,118 @@
+
+func (b *BackupTestSuite) SetupSuite() {
+	b.session = session.NewSession()
+
+	client, err := rancher.NewClient("", b.session)
+	require.NoError(b.T(), err)
+
+	b.client = client
+
+	subSession := b.session.NewSession()
+	defer subSession.Cleanup()
+}
+
+func (b *BackupTestSuite) TestS3InPlaceRestore() {
+	subSession := b.session.NewSession()
+	defer subSession.Cleanup()
+
+	project, err := projects.GetProjectByName(b.client, cluster, "System")
+	require.NoError(b.T(), err)
+
+	config.LoadConfig(charts.BackupRestoreConfigurationFileKey, backupRestoreConfig)
+
+	logrus.Info("Checking if the backup chart is already installed...")
+	initialBackupChart, err := shepCharts.GetChartStatus(b.client, project.ClusterID, "cattle-resources-system", "rancher-backup")
+	require.NoError(b.T(), err)
+
+	if !initialBackupChart.IsAlreadyInstalled {
+		installBroChart(b.client)
+	}
+
+	b.client, err = b.client.ReLogin()
+	require.NoError(b.T(), err)
+
+	logrus.Info("Creating two users, projects, and role templates...")
+	userList, projList, roleList, err := createRancherResources(b.client, project.ClusterID, "cluster")
+	require.NoError(b.T(), err)
+
+	logrus.Info("Provisioning a downstream RKE1 cluster...")
+	rke1ClusterObj, _, rke1ClusterConfig, err := createDownstreamCluster(b.client, "RKE1")
+	require.NoError(b.T(), err)
+
+	logrus.Info("Provisioning a downstream RKE2 cluster...")
+	_, rke2SteveObj, rke2ClusterConfig, err := createDownstreamCluster(b.client, "RKE2")
+	require.NoError(b.T(), err)
+
+	logrus.Info("Creating a backup of the local cluster...")
+	s3StorageLocation := backupRestoreConfig.S3BucketName + "/" + backupRestoreConfig.S3FolderName
+	_, backupFileName, err := createBackup(b.client, s3StorageLocation)
+	require.NoError(b.T(), err)
+
+	logrus.Info("Validating backup file is in AWS S3...")
+	backupPresent, err := checkAWSS3Object(s3StorageLocation, backupFileName)
+	require.NoError(b.T(), err)
+	assert.True(b.T(), backupPresent)
+
+	logrus.Info("Creating two more users, projects, and role templates...")
+	userListPostBackup, projListPostBackup, roleListPostBackup, err := createRancherResources(b.client, project.ClusterID, "cluster")
+	require.NoError(b.T(), err)
+
+	logrus.Infof("Creating a restore using backup file: %v", backupFileName)
+	restoreTemplate := bv1.NewRestore("", "", setRestoreObject())
+	restoreTemplate.Spec.BackupFilename = backupFileName
+	createdRestore, err := b.client.Steve.SteveType(restoreSteveType).Create(restoreTemplate)
+	require.NoError(b.T(), err)
+
+	restoreObj, err := b.client.Steve.SteveType(restoreSteveType).ByID(createdRestore.ID)
+	require.NoError(b.T(), err)
+
+	charts.VerifyRestoreCompleted(b.client, restoreSteveType, restoreObj)
+
+	logrus.Info("Validating Rancher resources...")
+	err = verifyRancherResources(b.client, userList, projList, roleList)
+	require.NoError(b.T(), err)
+
+	err = verifyRancherResources(b.client, userListPostBackup, projListPostBackup, roleListPostBackup)
+	assert.Error(b.T(), err)
```

Comment:
since the verifyRancherResources is written to return the first error, this isn't validating that projectList and roleList are throwing errors too.